src/main/java/org/update4j/util/PropertyManager.java

```diff
@@ -407,7 +410,7 @@ private static String trySystemProperty(String key, boolean systemInError) {
 		if (value != null) {
 			if (PLACEHOLDER.matcher(value)
 							.find()) {
-				throw new IllegalStateException("System properties must not contain placeholders.");
+				throw new IllegalStateException("System properties must not contain placeholders. "+ key +" "+value);
```

Comment:
Yep, was left out by me. Maybe just add a comma between key and value. And maybe use colon after `placeholders`.

Reply:
Done on next check in